tests/phpunit/integration/Modules/Analytics_4Test.php

```diff
@@ -535,8 +520,15 @@ public function test_get_report( $access_token ) {
 			$this->analytics->get_scopes()
 		);
 
-		$http_client = $this->create_fake_http_client( $property_id );
-		$this->analytics->get_client()->setHttpClient( $http_client );
+		$fake_http_handler = $this->create_fake_http_handler( $property_id );
+		FakeHttp::fake_google_http_handler(
+			$this->authentication->get_oauth_client()->get_client(),
+			$fake_http_handler
+		);
+		FakeHttp::fake_google_http_handler(
+			$this->analytics->get_owner_oauth_client()->get_client(),
+			$fake_http_handler
+		);
```

Comment:
This is a slight departure from the existing logic, where just setting the handler for `$this->analytics->get_client()` as below would be a direct mapping of it. I seem to recall you mentioned something about this, though, where we should have been updating the authentication client as well as the Analytics client. Therefore, I'm pretty confident your update here will be the right thing to do, but would you mind clarifying it for my understanding?

```php
		FakeHttp::fake_google_http_handler(
			$this->analytics->get_client(),
			$fake_http_handler
		);
```

Reply:
Ah, maybe it was [this comment](https://github.com/google/site-kit-wp/pull/6581#discussion_r1103512495) I was thinking of, which I didn't spot at the time of writing the above...

Reply:
Yes, I mentioned this above in https://github.com/google/site-kit-wp/pull/6581/files/b58274c240118142014bca49d98ad01c00d754a8#r1103512495.

I thought it was missing the handler set on the module owners client instance since the test uses the `data_access_token` data provider, which includes the empty token, so I probably assumed the test was intended to run in a sharing context as well, but this one is not so it's all good. I've restored the original form 👍